### PR TITLE
Wildcard match for media types

### PR DESCRIPTION
**Pull request by afallou**
**Why?**

A single resource schema can be broadcast using different media
types. For example different lifecycle changes could be broadcast to
different consumers, even though the schema is the same, e.g.
`application/vnd.foo.person.created` and
`application/vnd.foo.person.updated`. In those cases we only want to
register a single schema, and have it match for more than one media type.

**What?**

Allow registering a wildcard schema that matches on several
"concrete" schemas.
- Update registry `find` method.
- Add tests.
- Merge tags from `master` back to `develop`.

### CODE REVIEW

File: .bumpversion.cfg [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [bumpversion]
-current_version = 2.10.0
+current_version = 2.11.1
```
Comment (afallou) [inline .bumpversion.cfg]: I also took this opportunity to merge `master` back to `develop`.

Comment (cdm-ium) [reply .bumpversion.cfg]: Thank you.

File: microcosm_pubsub/registry.py [hunk 2]
```diff
@@ -72,7 +93,7 @@ def find(self, media_type):
 
         try:
             # use a concrete schema class if any
-            schema_cls = self._mappings[media_type]
+            schema_cls = self._mappings[matching_media_type]
```
Comment (cdm-ium) [inline microcosm_pubsub/registry.py]: Will this work properly with prefixes? I think you are potentialy returning a prefix on line 74?

Comment (afallou) [reply microcosm_pubsub/registry.py]: The assumption here is that the media type that's registered in `self._mappings` the library is either the full media type or the wildcard media type (`application/vnd.foo.*`). So when I pass in `applocation/vnd.foo.bar` I should first "translate" it to the registered wildcard, and only the wildcard from there on.

File: microcosm_pubsub/tests/test_registry.py [hunk 3]
```diff
@@ -86,6 +92,18 @@ def test_find(self):
...
+    def test_find_wildcard(self):
+        schema = self.registry.find("application/vnd.microcosm.wildcard.foo_12")
+        assert_that(schema, is_(instance_of(PubSubMessageCodec)))
+        assert_that(schema.schema, is_(instance_of(WildcardSchema)))
+
+    def test_wildcard_does_not_match_extended(self):
+        # The postfix `*` wildcard doesn't match on `.` separators
+        assert_that(
+            calling(self.registry.find).with_args("application/vnd.microcosm.wildcard.foo.bar"),
+            raises(KeyError),
```
Comment (rh-glob) [inline microcosm_pubsub/tests/test_registry.py]: Is this the `raise KeyError("Unregistered media type: {}".format(media_type))` case that is raised?

Comment (afallou) [reply microcosm_pubsub/tests/test_registry.py]: Yeah that's the one.

**Pull request merged**